PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/DirectoryAdapter.java

```diff
@@ -110,13 +112,24 @@ public void onClick(View v) {
                 SharedPreferences.Editor editedPreferences = sharedPref.edit();
                 ToggleButton star = (ToggleButton) v;
                 boolean starred = star.isChecked();
-                String currentName = currentPerson.getName();
+                String currentName = currentPerson.name;
```

Comment:
OK then, just change line 118 (`starredContacts.add(currentName);`) to `starredContacts.add(currentPerson.name);`